integration-tests/gatsby-image/cypress/integration/fixed.js

```diff
@@ -0,0 +1,53 @@
+const fixedTestId = `image-fixed`
+
+describe(`fixed`, () => {
+  beforeEach(() => {
+    cy.visit(`/`).waitForRouteChange()
+  })
+
+  it(`does not render a spacer div`, () => {
+    cy.getTestElement(fixedTestId)
+      .find(`.gatsby-image-outer-wrapper > .gatsby-image-wrapper > div`)
+      .should(`not.exist`)
+  })
+
+  it(`applies height and width to wrapper`, () => {
+    cy.getTestElement(fixedTestId)
+      .find(`.gatsby-image-wrapper`)
+      .should(`have.attr`, `style`)
+      .and(style => {
+        ;[`height:`, `width:`].forEach(part => {
+          expect(style).contains(part)
+        })
+      })
+  })
+
+  // TODO: figure out why these tests are failing
```

Comment:
how does this test fail?

Reply:
It's weird. It doesn't seem like `scrollIntoView` triggers the intersection observer, so the `picture` element is never added to the DOM/image.

So basically it times out trying to find a (non-existant) picture element. If you manually scroll in the cypress window when running `cy:open` the test works, so that's fun.

I fixed this by breaking apart the tests into individual pages, and then navigating to those pages. I would like to test the intersection observer behavior though!

Reply:
hmm yeah, this is weird.

Perhaps the tests are running too fast after scrolling? I.e. React hasn't had a chance to re-render and add in the DOM?

Reply:
Yeah, even with a timeout of like 10 seconds, still no dice. Cypress has pretty cool behavior here... so by default it will wait for 4 seconds for the element to appear, and then fail, otherwise you can override it.

So yeah, it seems like something isn't triggering, because as far as Cypress is concerned, the element isn't in the DOM.